File: webapi/Models/elimination/EliminationVariable.cs
using webapi.Models.Math;

namespace webapi.Models.Elimination
{
    public class EliminationVariable
    {
        public required string Name { get; set; }

        public required Coefficient Solution { get; set; }

        public required OptimalEquation OptimalEquation { get; set; }

        public EliminationVariable() { }
    }

    public enum OptimalEquation
    {
        First,
        Second,
        Both
    }
}

File: reactapp/src/types/elimination/eliminationVariable.ts
import { OptimalEquation } from "@/types/elimination/enums.ts";
import { Coefficient } from "@/types/math/coefficient.ts";

export interface IEliminationVariable {
    readonly name: string;
    readonly solution: Coefficient;
    readonly optimalEquation: OptimalEquation;
}

export class EliminationVariable implements IEliminationVariable {
    constructor(
        public readonly name: string,
        public readonly solution: Coefficient,
        public readonly optimalEquation: OptimalEquation
    ) {}
}
